app/src/main/java/org/p2p/wallet/newsend/NewSendPresenter.kt

```diff
@@ -71,14 +72,19 @@ class NewSendPresenter(
 
     override fun attach(view: NewSendContract.View) {
         super.attach(view)
-        initialize(view)
+        token?.let {
+            view.showToken(it)
+            calculationMode.updateLabels()
```

Comment:
you can just call `calculationMode.updateToken(it)` it will trigger `updateLabels` automatically